### ISSUE TITLE
--incremental --check-unsat-cores fatal failure at CVC4/src/./proof/sat_proof_implementation.h:803

### ISSUE THREAD

**Issue by muchang** [cvc5/cvc5#3651]
Hi,
For this formula,
```
(declare-fun a () Real)
(declare-fun b () Real)
(declare-fun c () Real)
(assert (> (+ (/ (- 2) a) (* 8 a)) 0))
(assert (<= b 0))
(assert (= (+ (* 4 a) (* 2 b)) 1))
(assert (>= (* b c) 0))
(check-sat)
(check-sat)
(check-sat)
(push)
(check-sat)
(pop)
(assert (>= c 1))
(check-sat)
```
`cvc4 --incremental --check-unsat-cores` throws out a fatal failure:
```
unknown
unknown
unknown
unknown
Fatal failure within CVC4::ClauseId CVC4::TSatProof<Solver>::resolveUnit(typename Solver::TLit) [with Solver = CVC4::Minisat::Solver; CVC4::ClauseId = unsigned int; typename Solver::TLit = CVC4::Minisat::Lit] at CVC4/src/./proof/sat_proof_implementation.h:803
Check failure
 reason_ref != Solver::TCRef_Undef
Aborted
```

OS: Ubuntu 18.04
Revision: dd29958

**Comment by HanielB** [comment]
As an internal note, this is works on proof-new. When we start using it for unsat-cores we will close this issue.

**Comment by HanielB** [comment]
Closed via #6574
